UI/Pages/Enquiry/Build/SchoolPostcode.cshtml.cs

```diff
@@ -0,0 +1,52 @@
+using Application.Common.Interfaces;
+using Application.Common.Models.Enquiry.Build;
+using Domain;
+
+namespace UI.Pages.Enquiry.Build;
+
+public class SchoolPostcode : PageModel
+{
+    private readonly ISessionService _sessionService;
+    private readonly IMediator _mediator;
+
+    public SchoolPostcode(ISessionService sessionService, IMediator mediator)
+    {
+        _sessionService = sessionService;
+        _mediator = mediator;
+    }
+    [BindProperty] public SchoolPostcodeModel Data { get; set; } = new();
+
+    public async Task<IActionResult> OnGetAsync(SchoolPostcodeModel data)
+    {
+        Data = data;
+
+        if (!string.IsNullOrWhiteSpace(Data.Postcode) && string.IsNullOrWhiteSpace(Data.SchoolPostcode))
+        {
+            var locationResult = await _mediator.Send(new GetSearchLocationQuery(Data.Postcode));
+            if (locationResult.TryValidate().IsSuccess)
+            {
+                Data.SchoolPostcode = Data.Postcode;
+            }
+        }
+
```

Comment:
Should we include `ModelState.Clear()`? We have already added it for other pages like `EnquirerEmail`. If we don't clear the ModelState, an error will be displayed in the page title.

Reply:
Added